src/card_flows/card_flow_delete_wallet.c

```diff
@@ -138,61 +149,83 @@ static bool check_wallet_already_deleted_on_card(
 card_error_type_e card_flow_delete_wallet(Wallet *selected_wallet) {
-  card_delete_share_cfg_t cfg = {.wallet = selected_wallet, .card_number = 0};
+  card_delete_share_cfg_t delete_cfg = {.wallet = selected_wallet,
+                                        .card_number = 0};
   card_error_type_e error_code = 0;
 
-  card_fetch_share_config_t configuration = {0};
+  card_fetch_share_config_t fetch_cfg = {0};
   card_fetch_share_response_t response = {0};
   char heading[MAX_HEADING_LEN] = "";
-  configuration.xcor = 0;
-  configuration.operation.expected_family_id = get_family_id();
-  configuration.frontend.msg = ui_text_place_card_below;
-  configuration.frontend.heading = heading;
-  configuration.frontend.unexpected_card_error = ui_text_wrong_card_sequence;
-  configuration.operation.skip_card_removal = true;
-  configuration.operation.buzzer_on_success = false;
+  fetch_cfg.xcor = 0;
+  fetch_cfg.operation.expected_family_id = get_family_id();
+  fetch_cfg.frontend.msg = ui_text_place_card_below;
+  fetch_cfg.frontend.heading = heading;
+  fetch_cfg.frontend.unexpected_card_error = ui_text_wrong_card_sequence;
+  fetch_cfg.operation.skip_card_removal = true;
+  fetch_cfg.operation.buzzer_on_success = false;
   response.card_info.tapped_family_id = NULL;
 
   for (int i = 1; i <= 4; i++) {
     snprintf(heading, MAX_HEADING_LEN, UI_TEXT_TAP_CARD, i);
-    configuration.operation.acceptable_cards = encode_card_number(i);
+    fetch_cfg.operation.acceptable_cards = encode_card_number(i);
 
     error_code = CARD_OPERATION_DEFAULT_INVALID;
-    cfg.card_number = i;
+    delete_cfg.card_number = i;
 
     // If wallet already deleted from ith card, skip card tapping
-    if (true == check_wallet_already_deleted_on_card(&cfg)) {
+    if (true == check_wallet_already_deleted_on_card(&delete_cfg)) {
       error_code = CARD_OPERATION_SUCCESS;
       continue;
     }
 
-    error_code = card_fetch_share(&configuration, &response);
+    error_code = card_fetch_share(&fetch_cfg, &response);
 
     if (CARD_OPERATION_SUCCESS != error_code) {
-      if (CARD_OPERATION_ABORT_OPERATION == error_code &&
-          SW_RECORD_NOT_FOUND == response.card_info.status) {
-        // In case wallet is not found on card, consider it as a success case as
-        // wallet is already deleted or not created on the card.
+      if ((SW_RECORD_NOT_FOUND == response.card_info.status) &&
+          (CARD_OPERATION_ABORT_OPERATION == error_code ||
+           CARD_OPERATION_VERIFICATION_FAILED == error_code)) {
+        // In case wallet is not found on card or wallet verification fails,
+        // consider it as a success case as wallet is already deleted, not
+        // created on card or wallet on card does not match with the required
+        // verification.
         clear_core_error_screen();
+        handle_wallet_deleted_from_card(&delete_cfg);
+        buzzer_start(BUZZER_DURATION);
+        continue;
```

Comment:
please add a note as to why we cannot afford to wait for card removal.

Reply:
updated, also added other changes.